python/rmm/_lib/memory_resource.pyx

```diff
@@ -236,24 +263,49 @@ cdef class LoggingResourceAdaptor(MemoryResource):
     cpdef flush(self):
         (<logging_resource_adaptor_wrapper*>(self.c_obj.get()))[0].flush()
 
+    cpdef get_file_name(self):
+        return self._log_file_name
+
+# Global per-device memory resources; dict of int:MemoryResource
+cdef dict _per_device_mrs = {}
+
+
+cpdef get_current_device():
```

Comment:
Since this is a cpdef function that returns an `int` which is a non-object type, we need to declare a value to use for exceptions (or we could use `except *` to always check if there's no return value that we could safely use)
```suggestion
cpdef int get_current_device() except -1:
```

Reply:
`Exception clause not allowed for function returning Python object`

Reply:
Missed `int` right after `cpdef`?